nodes/scene/timer.py

```diff
@@ -0,0 +1,391 @@
+                      ("STARTED", "Started", "", 2),
+                      ("PAUSED", "Paused", "", 3)]
+
+
+class SvTimerPropertyGroup(bpy.types.PropertyGroup):
+    ''' Timer class '''
+    timer_id: IntProperty(
+        name="Timer ID",
+        description="ID of the timer entry",
+        default=0)
+
+    timer_status: EnumProperty(
+        name="Timer Status", description="Timer Status",
+        items=timer_status_items,
+        default="STOPPED")
+
+    timer_time: FloatProperty(
+        name="Time", description="Elapsed time",
+        default=0.0)
+
+    timer_duration: FloatProperty(
+        name="Duration", description="Time after which the timer expires",
+        default=10.0)
+
+    timer_speed: FloatProperty(
+        name="Speed", description="Time speed as a multiplier of playback time",
+        default=1.0)
+
+    timer_last_operation: IntProperty(
+        name="Last Operation", description="Last external control operation executed",
+        default=0)
+
+    def id(self):
+        return self.timer_id
+
+    def status(self):
+        return self.timer_status
+
+    def elapsed_time(self, normalized=False):
+        if normalized:
+            return self.timer_time / self.timer_duration
+        else:
+            return self.timer_time
+
+    def remaining_time(self, normalized=False):
+        remaining_time = max(0, self.timer_duration - self.timer_time)
+        if normalized:
+            return remaining_time / self.timer_duration
+        else:
+            return remaining_time
+
+    def expired(self):
+        return self.timer_time >= self.timer_duration
+
+    def start(self):
+        print("* Timer {0}: START".format(self.timer_id))
+        if self.timer_status == "STOPPED":
+            self.timer_time = 0
+        self.timer_status = "STARTED"
+
+    def stop(self):
+        print("* Timer {0}: STOP".format(self.timer_id))
+        self.timer_status = "STOPPED"
+        self.timer_time = 0
+
+    def pause(self):
+        print("* Timer {0}: PAUSE".format(self.timer_id))
+        self.timer_status = "PAUSED"
+
+    def reset(self):
+        print("* Timer {0}: RESET".format(self.timer_id))
+        self.timer_time = 0
+
+    def expire(self):
+        print("* Timer {0}: EXPIRE".format(self.timer_id))
+        self.timer_time = self.timer_duration
+
+    def control_operation(self, operation):
+        if self.timer_last_operation != operation:
+            self.timer_last_operation = operation
+
+            if operation == 0:  # STOP timer
+                self.stop()
+
+            elif operation == 1:  # START/RESUME timer
+                self.start()
+
+            elif operation == 2:  # PAUSE timer
+                self.pause()
+
+            elif operation == 3:  # RESET timer
+                self.reset()
+
+            elif operation == 4:  # EXPIRE timer
+                self.expire()
+
+    def update(self, speed, duration, operation, lapse, loop):
```

Comment:
to avoid confusion please avoid naming a method just "update", maybe `update_config` ? or something else more to your liking.